File: subprojects/frida-python/examples/channels.py
import frida

device = frida.get_usb_device()

channel = device.open_channel("tcp:21")
print("Got channel:", channel)

welcome = channel.read(512)
print("Got welcome message:", welcome)

channel.write_all(b"CWD foo")
reply = channel.read(512)
print("Got reply:", reply)

channel.close()
print("Channel now:", channel)

File: subprojects/frida-node/examples/channels.ts
import frida from "frida";
import { inspect } from "util";

async function main() {
    const device = await frida.getUsbDevice();

    console.log("Getting channel...");
    // const channel = await device.openChannel("tcp:1234");
    const channel = await device.openChannel("lockdown:com.apple.instruments.remoteserver");
    console.log("Got channel:", inspect(channel, {
        colors: true,
        breakLength: Infinity
    }));
}

main()
    .catch(e => {
        console.error(e.stack);
    });
